service/tablet_allocator.cc

```diff
@@ -1709,6 +1738,210 @@ class load_balancer {
         co_return std::move(plan);
     }
 
+    future<migration_plan> make_global_rebalancing_plan(dc_name dc, node_load_map& nodes) {
+        migration_plan plan;
+
+        mutable_token_metadata_ptr temp_tm = make_lw_shared<token_metadata>(co_await _tm->clone_async());
+        temp_tm->set_tablets(tablet_metadata());
+
+        // Need stable table order so that the next call to make_global_rebalancing_plan() converges
+        // to the same distribution.
+        std::set<table_id> tables;
+        for (auto&& [table, load] : _tablet_count_per_table) {
+            tables.insert(table);
+        }
+
+        for (auto table : tables) {
+            auto s = _db.find_schema(table);
+            auto& rs = _db.find_keyspace(s->ks_name()).get_replication_strategy();
+            auto& old_map = _tm->tablets().get_tablet_map(s->id());
+            auto tablet_count = old_map.tablet_count();
+
+            lblogger.trace("Allocating {} tablets for table {}", tablet_count, table);
+
+            auto map = co_await rs.maybe_as_tablet_aware()->allocate_tablets_for_new_table(s, temp_tm, 1, tablet_count);
+
```

Comment:
This for sure will cause unnecessary movement.

I think the goal layout should start as a derivative of the current layout:

1. Clone the current layout
2. move tablets around to reach goal (without moving actual data)
3. Have real layout chase goal layout
4. Incrementally update goal layout to reflect changes (don't generate it anew each time)